### PR TITLE
Azure Toolkit for Rider: Deployment Slot setting is not saved #423

### LINKED ISSUE #423: Azure Toolkit for Rider: Deployment Slot setting is not saved
JetBrains Rider 2020.3
Build #RD-203.5981.141, built on December 11, 2020
Runtime version: 11.0.9.1+11-b1145.24 amd64
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.
.NET Framework 4.0.30319.42000
Windows 10 10.0
GC: ParNew, ConcurrentMarkSweep
Memory: 1450M
Cores: 16
Registry: ide.tree.horizontal.default.autoscrolling=false, performance.watcher.sampling.interval.ms=200, ide.tooltip.showAllSeverities=true, performance.watcher.unresponsive.interval.ms=1000, ide.tree.expand.navigatable.on.double.click.disabled=true, ide.tooltip.initialDelay.highlighter=0, search.everywhere.settings=true, parameter.info.max.visible.rows=10, ide.win.file.chooser.native=true, search.everywhere.pattern.checking=false, ide.tooltip.initialDelay=0, ide.balloon.shadow.size=0, ide.require.transaction.for.model.changes=false, js.debugger.webconsole=false, ide.new.project.model=false, rdclient.asyncActions=false
Non-Bundled Plugins: com.jetbrains.rider.fsharp, com.intellij.resharper.azure

Plugin version: 3.46.0.1046-2020.3

I tried to publish my app to an azure app service slot, which succeeds on first try. After some time I used the same configuration to re-deploy the app, but the configuration management "forgot" the deployment slot setting, resulting in the app being deployed to the production slot. When I edit the publish profile and select the correct deployment slot, click on Apply/Ok and reopen the configuration management, the deployment slot is unselected again.

### PR DESCRIPTION
**Pull request by maartenba**
Fixes #423

1) Slot settings were not displayed correctly when editing configuration
2) After publish, app id was overwritten by the slot id (added tests for these cases)

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/ui/component/appservice/DeploymentSlotPublishSettingsPanel.kt [hunk 1]
```diff
@@ -106,6 +110,19 @@ abstract class DeploymentSlotPublishSettingsPanelBase<T: DeploymentSlotBase<T>>(
             checkBoxIsEnabled.isEnabled = slots.isNotEmpty()
             this.isEnabled = slots.isNotEmpty()
 
+            initialData.value?.let { init ->
```
Comment (sdubov) [inline PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/ui/component/appservice/DeploymentSlotPublishSettingsPanel.kt]: @maartenba, just want to discuss - would not it be more clear for code readers to fire an event here about loadFinished and subscribe on this event in `resetFromConfig()` once (via `adviceOnce`) to make sure we reset this only once when we load slots? Then, I believe, you would not need to wrap the result into `InitialDeploymentSlotData` and just handle the event properly inside a component that carry `resetFromConfig()` call itself.

**Review by sdubov — COMMENTED** [review]
I've checked the change and it works. One small question about events logic.

File: PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/ui/component/appservice/DeploymentSlotPublishSettingsPanel.kt (hunk 1 above)
Comment (maartenba) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/ui/component/appservice/DeploymentSlotPublishSettingsPanel.kt]: Yes! Let me fix that.

Comment (maartenba) [reply PluginsAndFeatures/azure-toolkit-for-intellij/rider/src/com/microsoft/intellij/ui/component/appservice/DeploymentSlotPublishSettingsPanel.kt]: There you go :)

**Review by sdubov — APPROVED** [review]

**Pull request merged**